machineconfig/master/spec/config/storage/files/haproxy.sh L3Zhci91c3Jsb2NhbC9iaW4vaGFwcm94eS5zaAo=

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+set -e
+
+
+function generate_haproxy_cfg {
+envsubst < $HOST_CFG_PATH/haproxy.cfg.template | sudo tee $HOST_CFG_PATH/haproxy.cfg
+
+etcd_members=$(dig +noall +answer -t SRV _etcd-server-ssl._tcp.$CLUSTER_DOMAIN | awk '{print $NF}')
+for item in $etcd_members
+do
+    ips=$(dig +noall +answer $item  | awk '{print $NF}')
+    if [[ "$?" -eq 0 && "${#ips[@]}" -ne 0 ]]; then
+        sudo echo "   server "$item" "$ips":"$OCP_API_PORT" weight 1 check inter 3s fall 3 rise 3" >>  $HOST_CFG_PATH/haproxy.cfg
+    fi
+done
+
+}
+
+function does_lb_topology_changed {
+etcd_members=$(dig +noall +answer -t SRV _etcd-server-ssl._tcp.$CLUSTER_DOMAIN   | awk '{print $NF}')
+for item in $etcd_members
+do
+    ips=$(dig +noall +answer $item | awk '{print $NF}')
+    haproxy_cfg_ip=($(grep $item $HOST_CFG_PATH/haproxy.cfg | awk '{print $3}' | sed 's/:.*//'))
+    if [[ "$?" -eq 0 && "${#ips[@]}" -ne 0 ]]; then
+       if [ "$haproxy_cfg_ip" != "$ips" ]; then
+           return 0
+       fi
+    else
+       if [[! -z "${haproxy_cfg_ip// }" ]];then
+          return 0
+       fi
+    fi
+done
+return 1
+}
+
+function start_haproxy {
+sudo mkdir --parents $HOST_CFG_PATH
+
+generate_haproxy_cfg
+
+if ! podman inspect "$HAPROXY_IMAGE" &>/dev/null; then
+    echo "Pulling release image..."
+    podman pull "$HAPROXY_IMAGE"
+fi
+
+sudo podman run \
+            -d \
+            --name masters-haproxy \
+            --network=host \
+            --cap-add=NET_ADMIN \
+            -v $HOST_CFG_PATH:/usr/local/etc/haproxy:ro \
+            $HAPROXY_IMAGE
+
+# redirect traffic to OCP API LB
+sudo iptables -t nat -I PREROUTING --src 0/0 --dst $API_VIP -p tcp --dport $OCP_API_PORT -j REDIRECT --to-ports $LB_OCP_API_PORT -m comment --comment "OCP_API_LB_REDIRECT"
```

Comment:
Shouldn't you check the rule is already there and if that's the case, replace it?

Reply:
OH, I see you made a stop function